rhino/src/main/java/org/mozilla/javascript/Context.java

```diff
@@ -1749,7 +1749,8 @@ public static Object javaToJS(Object value, Scriptable scope, Context cx) {
         if (value instanceof String
                 || value instanceof Number
                 || value instanceof Boolean
-                || value instanceof Scriptable) {
+                || value instanceof Scriptable
+                || value instanceof Undefined) {
```

Comment:
do not pass undefined to wrapperFactory

Reply:
Can you elaborate on the 'why' of this?

Reply:
`Undefined` is already a js compatible object, so I think, it does not need a `javaToJs` conversion.

Background: I plan to make the wrapperFactory optional (no wrapperFactory == no liveConnect) and I found some failing tests, that have passed `undefined` here, although they do not use use liveConnect

> If value is an instance of String, Number, Boolean, Function or Scriptable, it is returned
> as it and will be treated as the corresponding JavaScript type of string, number, boolean,
> function and object.

I will line up the code with the documentation. (Function mentioned in doc, but not handled in code) and maybe extract this in a separate commit.